mne/encoding/model.py

```diff
@@ -0,0 +1,230 @@
+            raise ValueError('condition must be one of "all" or "any"')
+        self.mask_condition = mask_condition
+        self.est = _check_estimator(estimator)
+
+    def fit(self, X, y):
+        """Remove datapoints then fit the estimator.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples, n_features)
+            The input array for the model. Rows will be removed.
+        y : array, shape (n_samples, n_targets)
+            The output array for the model. Rows will be removed.
+        """
+        X, y = self.mask_data(X, y, ixs=self.ixs)
+        self.est.fit(X, y)
+        return self
+
+    def predict(self, X):
+        """Call self.est.predict.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples, n_features)
+            The input array for the model. Rows will NOT be removed.
+        """
+        X = self.mask_data(X, ixs=self.ixs_pred)
+        return self.est.predict(X)
+
+    def mask_data(self, X, y=None, ixs=None):
+        """Remove datapoints according to indices or masked values.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples, n_features)
+            The input array for the model. Rows will be removed.
+        y : array, shape (n_samples, n_targets)
+            The output array for the model. Rows will be removed.
+        ixs : array, shape (n_samples_to_keep,)
+            An optional array of indices to keep before removing
+            any samples based on their values.
+
+        Returns
+        -------
+        X : array, shape (n_samples_after_removal, n_features)
+            The input X array after removing rows based on ixs, then
+            based on the values in X.
+        y : array, shape (n_samples_after_removal, n_targets) (optional)
+            The output y array after removing rows based on ixs, then
+            based on the values in X. Only returned if `y` is passed.
+        """
+
+        # First remove indices if we've manually specified them
+        if y is not None:
+            if y.ndim == 1:
+                y = y[:, np.newaxis]
+        if ixs is not None:
+            if X.shape[0] < ixs.max():
+                raise ValueError('ixs exceed data shape')
+            X = X[ixs]
+            if y is not None:
+                y = y[ixs]
+
+        # Find data points with value we want to remove
+        if self.mask_val is not None:
+            if hasattr(self.mask_val, '__call__'):
+                # If a callable, pass X as an input, assume bool output
+                mask = self.mask_val(X)
+                if mask.ndim > 1:
+                    raise ValueError('Output mask must be shape (n_samples,)')
+            else:
+                if self.mask_val is np.nan:
+                    mask = ~np.isnan(X)
+                else:
+                    mask = X != self.mask_val
+
+                # Decide which rows to remove
+                if self.mask_condition == 'all':
```

Comment:
I believe here I was keeping it this way because using `np.all` was breaking with sparse matrices.